lib/std/os/uefi/protocol/file.zig

```diff
@@ -40,108 +94,195 @@ pub const File = extern struct {
+        buffer: []const u8,
+    ) SetInfoError!void {
+        switch (self._set_info(self, information_type, buffer.len, buffer.ptr)) {
+            .success => {},
+            .unsupported => return Error.Unsupported,
+            .no_media => return Error.NoMedia,
+            .device_error => return Error.DeviceError,
+            .volume_corrupted => return Error.VolumeCorrupted,
+            .write_protected => return Error.WriteProtected,
+            .access_denied => return Error.AccessDenied,
+            .volume_full => return Error.VolumeFull,
+            .bad_buffer_size => return Error.BadBufferSize,
+            else => |status| return uefi.unexpectedStatus(status),
+        }
     }
 
-    pub fn flush(self: *const File) Status {
-        return self._flush(self);
+    pub fn flush(self: *File) FlushError!void {
+        switch (self._flush(self)) {
+            .success => {},
+            .device_error => return Error.DeviceError,
+            .volume_corrupted => return Error.VolumeCorrupted,
+            .write_protected => return Error.WriteProtected,
+            .access_denied => return Error.AccessDenied,
+            .volume_full => return Error.VolumeFull,
+            else => |status| return uefi.unexpectedStatus(status),
+        }
     }
 
-    pub const efi_file_mode_read: u64 = 0x0000000000000001;
-    pub const efi_file_mode_write: u64 = 0x0000000000000002;
-    pub const efi_file_mode_create: u64 = 0x8000000000000000;
+    pub const OpenMode = enum(u64) {
+        read = 0x0000000000000001,
+        // implies read
+        write = 0x0000000000000002,
+        // implies read+write
+        create = 0x8000000000000000,
+    };
+
+    pub const Attributes = packed struct(u64) {
+        // 0x0000000000000001
+        read_only: bool = false,
+        // 0x0000000000000002
+        hidden: bool = false,
+        // 0x0000000000000004
+        system: bool = false,
+        // 0x0000000000000008
+        reserved: bool = false,
+        // 0x0000000000000010
+        directory: bool = false,
+        // 0x0000000000000020
+        archive: bool = false,
+        // used exclusively for `valid_attr` as far as i can tell...
+        _flag: bool = false,
+        _pad: u57 = 0,
 
-    pub const efi_file_read_only: u64 = 0x0000000000000001;
-    pub const efi_file_hidden: u64 = 0x0000000000000002;
-    pub const efi_file_system: u64 = 0x0000000000000004;
-    pub const efi_file_reserved: u64 = 0x0000000000000008;
-    pub const efi_file_directory: u64 = 0x0000000000000010;
-    pub const efi_file_archive: u64 = 0x0000000000000020;
-    pub const efi_file_valid_attr: u64 = 0x0000000000000037;
+        // 0x0000000000000037
+        pub const valid_attr: Attributes = .{
+            .read_only = true,
+            .system = true,
+            ._flag = true,
+        };
+    };
 
-    pub const efi_file_position_end_of_file: u64 = 0xffffffffffffffff;
+    const efi_file_position_end_of_file: u64 = 0xffffffffffffffff;
```

Comment:
Drop `efi_file_` (see https://ziglang.org/documentation/master/#Avoid-Redundancy-in-Names)